assets/basic/json.sublime-syntax

```diff
@@ -16,7 +16,7 @@ contexts:
         - meta_scope: comment.block.jsonkv
         - match: \*/
           pop: true
-    - match: '(")(?i)([^\"]+)(")\s*?:'
+    - match: '(")(?i)([^\\"]+)(")\s*?:'

```

Comment:
This would break when keys contain a backslash e.g `xh --offline : "hel\lo=world"`.

#### Before:
<img width="440" alt="Screenshot 2021-03-17 230840" src="https://user-images.githubusercontent.com/16341131/111550452-07b31680-8776-11eb-869b-6c894cfb1fd9.png">

#### After:
<img width="494" alt="Screenshot 2021-03-17 230622" src="https://user-images.githubusercontent.com/16341131/111550434-fbc75480-8775-11eb-98d4-630f8e891907.png">



Interestingly, I have noticed that `xh --offline : 'hel"\lo=world'` confuses the highlighter both in develop and and this PR

<img width="511" alt="Screenshot 2021-03-17 231002" src="https://user-images.githubusercontent.com/16341131/111550613-5c569180-8776-11eb-936b-4449e8351a01.png">

Reply:
Yep, the highlighting for keys is a bit unreliable. It also fails on empty strings.

It's not a severe problem, and I can't think of a fix I'd be confident in. So I think it's ok as is.

Later on I'd like to see if I can integrate the JSON highlighting with jsonxf. That should make it faster and more reliable.

Another fun case where the key becomes yellow is when highlighting this JSON in streaming mode:
```json
{"foo"
:"bar"}
```
It highlights one line at once, so it doesn't see the colon in time.

Reply:
I agree that color spill is worse than keys that become yellow or blue so I will approve this.

<img width="413" alt="Screenshot 2021-03-18 214226" src="https://user-images.githubusercontent.com/16341131/111701697-17446500-8833-11eb-9861-6a5557fc81d3.png">


By the way, when did we start sorting JSON keys?

Reply:
serde_json does that. Seems that we could turn it off by using the `preserve_order` feature.

It only happens for requests, not responses.

Reply:
Although the behavior would be different than HTTPie, I don't think it is a good idea to sort JSON keys.

Reply:
Agreed, the order in which you specify them is the most intuitive.

HTTPie doesn't sort them when sending either, just when printing. So keeping insertion order would arguably be closer to HTTPie's behavior.